src/wp-includes/block-patterns.php

```diff
@@ -328,6 +328,14 @@ function _register_remote_theme_patterns() {
  * @access private
  */
 function _register_theme_block_patterns() {
+
+	/**
+	 * Bail early if WP is installing, since the theme may not be fully loaded at this point.
+	 */
```

Comment:
Maybe `wp_using_themes()` would be a better check here. 

It returns false for:
* installation
* upgrade
* setup config

WP_INSTALL is active when hitting `wp-activate.php` for multisite activation, it is themed so I suspect wp_using_themes will return true but am currently confirming

```suggestion
	// Bail early if WP is installing, since the theme may not be fully loaded at this point.
```

CS Nit: single line comment style.

Reply:
Would it make sense to duplicate the conditions of the early return in `wp_get_active_and_valid_themes()` which is

```
if ( wp_installing() && 'wp-activate.php' !== $pagenow ) {
   return $themes;
}
```

Reply:
> Would it make sense to duplicate the conditions of the early return in `wp_get_active_and_valid_themes()` which is

Yeah, I think that will need to be the case.

On multisite the wp-activate.php entry point `wp_using_themes()` returns `false` but it does load at least some elements of themes, setting the colour in the site editors styles affects the font colour on the activation screen.

Reply:
After looking at this again, the safest option is to rely directly on `wp_get_active_and_valid_themes()`, since there are multiple parts of that logic that could result in an empty array being returned, causing no `functions.php` file to be loaded.

Updated in [324ae8d](https://github.com/WordPress/wordpress-develop/pull/5568/commits/324ae8d3f0f9da7be13dd0303724748501a04c4c). Tests added.